app/models/group.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+#
+# Model for Housing Groups
+#
+# @attr [Integer] size The room size that the Group wants.
+# @attr [String] status The state of the group (open, full, or locked)
+# @attr [User] leader The student that represents the Group.
+# @attr [Array<User>] members The members of the group, excluding the leader.
+# @attr [Draw] draw The Draw that this Group is in.
+class Group < ApplicationRecord
```

Comment:
this is missing the more complex validations (no more students than size, all students must be in the same draw, size must be available in the current draw, etc), but I think those should be done in a separate issue because this PR is already fairly long